src/jit/codegenxarch.cpp

```diff
@@ -5664,16 +5592,35 @@ void CodeGen::genCallInstruction(GenTreePtr node)
 
     // Determine return value size.
     emitAttr retSize = EA_PTRSIZE;
-    if (call->gtType == TYP_REF ||
-        call->gtType == TYP_ARRAY)
+    emitAttr retSize1 = EA_UNKNOWN;
+#if FEATURE_MULTIREG_RET
+    if (varTypeIsStruct(call->gtType))
     {
-        retSize = EA_GCREF;
+        // Make sure it is a multi-register passed struct,
+        // otherwise, for a struct passed in a single register 
+        // the call would have a normalized type that is not a struct type.
+        assert(call->structDesc.passedInRegisters &&
+               (call->structDesc.eightByteCount == CLR_SYSTEMV_MAX_EIGHTBYTES_COUNT_TO_RETURN_IN_REGISTERS));
+
+        retSize = EA_ATTR(emitTypeSize(compiler->getEightByteType(call->structDesc, 0)));
```

Comment:
Why is the EA_ATTR needed here? emitTypeSize returns an emitAttr. Also, I think it is better to use different names for emitAttr than "size", since emitAttr encodes more than just size. For example, emitAttrRet1 and emitAttrRet2.